src/main/java/codeu/controller/ActivityFeedServlet.java

```diff
@@ -1,25 +1,171 @@
-package codeu.controller;
-
-import java.io.IOException;
-
-import javax.servlet.ServletException;
-import javax.servlet.http.HttpServlet;
-import javax.servlet.http.HttpServletRequest;
-import javax.servlet.http.HttpServletResponse;
-
-/** Servlet class responsible for the activity feed page. */
-public class ActivityFeedServlet extends HttpServlet 
-{
-	
-	/**
-	 * This function fires when a user requests the /activity URL. It simply
-	 * forwards the request to activity.jsp.
-	 * @throws IOException 
-	 * @throws ServletException 
-	 */
-	@Override
-	public void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException 
-	{
-	    request.getRequestDispatcher("/WEB-INF/view/activityfeed.jsp").forward(request, response);
-	}
-}
+package codeu.controller;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.*;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import codeu.model.data.Conversation;
+import codeu.model.data.Message;
+import codeu.model.data.User;
+import codeu.model.store.basic.ConversationStore;
+import codeu.model.store.basic.MessageStore;
+import codeu.model.store.basic.UserStore;
+
+/** Servlet class responsible for the activity feed page. */
+public class ActivityFeedServlet extends HttpServlet {
+	/** Store class that gives access to Conversations. */
+	private ConversationStore conversationStore;
+
+	/** Store class that gives access to Messages. */
+	private MessageStore messageStore;
+
+	/** Store class that gives access to Users. */
+	private UserStore userStore;
+
+	ArrayList<Instant> arrInstantsSorted = new ArrayList<Instant>();

```

Comment:
I think servlets are accessed concurrently, so storing data in a local variable will cause races.